common/environment/environment.cpp

```diff
@@ -31,11 +31,77 @@
 #include "dalienv.hpp"
 
 #define SDS_LOCK_TIMEOUT  30000
-
+#define DEFAULT_DROPZONE_INDEX      1
+#define DROPZONE_BY_MACHINE_SUFFIX  "-dropzoneByMachine-"
+#define DROPZONE_SUFFIX             "dropzone-"
+#define MACHINE_PREFIX              "machine-"
+#define NUMBER_OF_DROPZONES_SUFFIX  "-numberOfDropZones"
```

Comment:
This approach works, by mangling name by count+index+machine etc. and adding them to std. cache.
And necessary if using the keyed 'cache' as it stand, but it makes the other code more complicated than it would otherwise be.
And since you require a utility 'numOfDropzonesByComputer' property tree for this to be maintained, I think you'd be better to simplify and use a IConstDropZoneInfo cache/table.